bundles/org.eclipse.passage.lic.base/src/org/eclipse/passage/lic/base/requirements/LicensingRequirements.java

```diff
@@ -44,7 +45,8 @@ private LicensingRequirements() {
 	public static LicensingRequirement extractFromCapability(String bundleName, String bundleVendor,
 			Map<String, Object> attributes, Map<String, String> directives, Object source) {
 		if (attributes == null) {
```

Comment:
For the cases where nullability is unavoidable because of foreign component's desing, can I offer you to use
```java
Optional.ofNullable(attributes)
```
?